src/Client.php

```diff
@@ -111,9 +123,10 @@ public function __construct(?string $uri = null, array $uriOptions = [], array $
         $driverOptions['driver'] = $this->mergeDriverInfo($driverOptions['driver'] ?? []);
 
         $this->uri = $uri ?? self::DEFAULT_URI;
+        $this->builderEncoder = $driverOptions['builderEncoder'] ?? new BuilderEncoder();
```

Comment:
Static analysis could be better at detecting the type here. Line 107, we assert the value is not set, `null` or an instance of `Encoder`.